Sources/SmileID/Classes/Helpers/LocalStorage.swift

```diff
@@ -20,169 +22,173 @@ public class LocalStorage {
         }
     }
 
-    static func saveImage(
-        image: Data,
-        to folder: String = "sid-\(UUID().uuidString)",
-        name: String
-    ) throws -> URL {
-        try createDefaultDirectory()
-        let destinationFolder = try defaultDirectory.appendingPathComponent(folder)
-        try createDirectory(at: destinationFolder, overwrite: false)
-        let fileName = filename(for: name)
-        return try write(image, to: destinationFolder.appendingPathComponent(fileName))
+    static var unsubmittedDirectory: URL {
+        get throws {
+            try defaultDirectory.appendingPathComponent(pendingFolderName)
+        }
     }
 
-    static func saveSelfieImages(
-        selfieImage: Data,
-        livenessImages: [Data],
-        to folder: String = "sid-\(UUID().uuidString)"
-    ) throws -> SelfieCaptureResultStore {
-        try createDefaultDirectory()
-        let destinationFolder = try defaultDirectory.appendingPathComponent(folder)
-        var livenessUrls = [URL]()
-        try createDirectory(at: destinationFolder, overwrite: false)
-        var imageInfoArray = try livenessImages.map { [self] imageData in
-            let fileName = filename(for: "liveness")
-            let url = try write(imageData, to: destinationFolder.appendingPathComponent(fileName))
-            livenessUrls.append(url)
-            return UploadImageInfo(imageTypeId: .livenessJpgFile, fileName: fileName)
+    static var submittedDirectory: URL {
+        get throws {
+            try defaultDirectory.appendingPathComponent(completedFolderName)
         }
-        let fileName = filename(for: "selfie")
-        let selfieUrl = try write(
-            selfieImage,
-            to: destinationFolder.appendingPathComponent(fileName)
-        )
-        imageInfoArray.append(UploadImageInfo(imageTypeId: .selfieJpgFile, fileName: fileName))
-        return SelfieCaptureResultStore(
-            selfie: selfieUrl,
-            livenessImages: livenessUrls
-        )
     }
 
-    static func createInfoJson(
-        selfie: URL,
-        livenessImages: [URL],
+    private static func createSmileFile(
+        to folder: String,
+        name: String,
+        file data: Data
+    ) throws -> URL {
+        try createDirectory(at: defaultDirectory, overwrite: false)
+        try createDirectory(at: unsubmittedDirectory, overwrite: false)
```

Comment:
Instead of 2 call to this function, I think we can just simplify our `createDirectory` function to just create the intermediate directories

```diff
-    private static func createDirectory(at url: URL, overwrite: Bool = true) throws {
-        if !fileManager.fileExists(atPath: url.relativePath) {
-            try FileManager.default.createDirectory(at: url, withIntermediateDirectories: false)
-        } else {
-            if overwrite {
-                try delete(at: url)
-                try createDirectory(at: url)
-            }
-        }
+    private static func createDirectory(at url: URL) throws {
+        try FileManager.default.createDirectory(at: url, withIntermediateDirectories: true)
    }
```

that allows us to consolidate this to just
```swift
try createDirectory(at: unsubmittedDirectory)
```

Reply:
Fixed